opensrp-reveal/src/test/java/org/smartregister/reveal/sync/RevealClientProcessorTest.java

```diff
@@ -214,6 +238,154 @@ public void testProcessEventClientShouldUpdateLocation() {
+        Client client = new Client(baseEntityId);
+        client.setFirstName("Jordan");
+        Address address = new Address();
+        Map<String, String> addressFields = new HashMap<>();
+        address.setAddressFields(addressFields);
+        client.setAddresses(Collections.singletonList(address));
+
+        when(structureRepository.getLocationById(event.getDetails().get(LOCATION_UUID))).thenReturn(structure);
+
+        clientProcessor = spy(clientProcessor);
+
+        clientProcessor.processClient(Collections.singletonList(new EventClient(event, client)), true);
+
+        verify(structureRepository).addOrUpdate(structureCaptor.capture());
+        assertEquals(client.getFirstName(), structureCaptor.getValue().getProperties().getCustomProperties().get(STRUCTURE_NAME));
+        assertNull(structureCaptor.getValue().getProperties().getName());
+        assertEquals(BaseRepository.TYPE_Unsynced, structureCaptor.getValue().getSyncStatus());
+
+        verify(clientProcessor).processEvent(eventArgumentCaptor.capture(), clientArgumentCaptor.capture(), any());
+        assertEquals(event.getEventId(), eventArgumentCaptor.getValue().getEventId());
+        assertEquals(event.getEventType(), eventArgumentCaptor.getValue().getEventType());
+        assertEquals(client.getBaseEntityId(), clientArgumentCaptor.getValue().getBaseEntityId());
+
+    }
+
+    @Test
+    public void testProcessRegisterStructureEvent() throws Exception {
+
+        String locationParent = UUID.randomUUID().toString();
+        String eventId = UUID.randomUUID().toString();
+        String baseEntityId = UUID.randomUUID().toString();
+        event.setBaseEntityId(baseEntityId);
+        event.getDetails().put(LOCATION_PARENT, locationParent);
+        event.setEventId(eventId);
+        event.setEventType(REGISTER_STRUCTURE_EVENT);
+        clientProcessor = spy(clientProcessor);
+
+        clientProcessor.processClient(Collections.singletonList(new EventClient(event, null)), true);
+
+        verify(clientProcessor).processEvent(eventArgumentCaptor.capture(), clientArgumentCaptor.capture(), any());
+        assertEquals(event.getEventId(), eventArgumentCaptor.getValue().getEventId());
+        assertEquals(event.getEventType(), eventArgumentCaptor.getValue().getEventType());
+        assertEquals(baseEntityId, clientArgumentCaptor.getValue().getBaseEntityId());
+
+    }
+
+    @Test
+    public void testProcessLarvalDippingEvent() throws Exception {
+
+        testLocationInterventionEvet(LARVAL_DIPPING_EVENT);
+    }
+
+    @Test
+    public void testProcessBednetDistributionEvent() throws Exception {
+
+        testLocationInterventionEvet(BEDNET_DISTRIBUTION_EVENT);
+    }
+
+    @Test
+    public void testProcessBehaviouralChangeEvent() throws Exception {
+
+        testLocationInterventionEvet(BEHAVIOUR_CHANGE_COMMUNICATION);
+    }
+
+    @Test
+    public void testProcessMosquitoCollectionEvent() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5066421)